p2p/protocol.py

```diff
@@ -116,6 +122,120 @@ def encode(self, data: _DecodedMsgType) -> Tuple[bytes, bytes]:
         return header, body
 
 
+class BaseRequest(ABC):
+    """
+    Base representation of a *request* to a connected peer which has a matching
+    *response*.
+    """
+    @abstractmethod
+    def validate_response(self, response: Any) -> None:
+        pass
+
+
+class BaseHeaderRequest(BaseRequest):
+    block_number_or_hash: BlockIdentifier
+    max_headers: int
+    skip: int
+    reverse: bool
+
+    @property
+    @abstractmethod
+    def MAX_HEADERS_FETCH(self) -> int:
+        pass
```

Comment:
shouldn't this raise NotImplementedError()? We definitely don't want it to return None, right?

Reply:
The `abstractmethod` decorator should take care of this for us, preventing us from instantiating the class if this property has not been overridden.